### ISSUE TITLE
Mistral workflow stuck in 'running'

### ISSUE THREAD

**Issue by dzimine** [StackStorm/st2#2123]
Sometimes Mistral workflow got stuck in "running". This is a corner case, we are fixing it in [Mistral upstream](https://bugs.launchpad.net/mistral/+bugs?orderby=-datecreated&start=0), when it happens, here's a way to unblock yourself. 

Before you start, please update the bug with 1) your workflow 2) errors from `/var/log/mistral.log`. 
### How to un-stuck.
1. cancel st2 execution, most time this helps. st2 execution cancel <execution_id>
2. still running? Ok, there's a mistral task which got stuck. **This is worse**. 
3. find out mistral workflow execution id. It's in workflow action output, under each task. `tasks[0].workflow_execution_id`
4. check to see mistral workflow has failed
   `mistral execution-list` or `mistral execution-get <mistral_workflow_execution_id>`
5. figure out which task is stuck. `mistral task-list` - look for RUNNING, with corresponding mistral_workflow_execution_id. Make a note on task execution id.
6. Update mistral DB to force the task to `ERROR`. 

```
    # look up the  credentials
    cat /etc/mistral/mistral.conf 
    # connect to DB
     psql -d mistral -h localhost -U mistral -W <password>
    # find the task execution
    select * from executions_v2 where id='eca7d802-3089-45e6-a651-504e3b5f18c9'
    # force it to be "ERROR"
    update executions_v2 set state='ERROR' where id='eca7d802-3089-45e6-a651-504e3b5f18c9'
```

**Comment by srimandaleeka01** [comment]
@dzimine Is there a mistral bug id we could track ? Thanks !!

**Comment by m4dcoder** [comment]
@srimandaleeka01 Are you experiencing this issue in the latest v2.4?

**Comment by srimandaleeka01** [comment]
@m4dcoder Yeah we are running into this issue. We are using v2.2.1 version of stack storm.

**Comment by m4dcoder** [comment]
@srimandaleeka01 Please upgrade to v2.4.0. This issue was fixed awhile back but then there was a regression due to some refactoring. The regression has been fixed in 2.4. Please refer to this issue at https://github.com/StackStorm/st2/issues/3680.

**Comment by srimandaleeka01** [comment]
Thanks @m4dcoder ! Appreciate the quick turn around .
